docs/CONTRIBUTING.md

```diff
@@ -46,13 +46,60 @@ make update-assets
 
 Commit and submit a PR.
 
-## Commit guidelines
-
-The title of the commit message should describe the _what_ about the
-changes. Additionally, it is helpful to add the _why_ in the body of
-the commit changes. Make sure to include any assumptions that you
-might have made in this commit. Changes to unrelated parts of the
-codebase should be kept as separate commits.
+## Contributing PRs
+
+This is a general guideline to make PRs/commits easier to review. Use your own
+judgement to decide when it is best to follow them. These are not strict rules
+but highly recommended to follow:
+
+ * Commits should be atomic and self-contained (in the sense of not breaking
+   tests, _logically_ different changes in different commits). This is best
+   explained in the the Linux Kernel [submitting patches][linux-sep-changes]
+   guide.
+
+ * Commit messages should explain the intention, _why_ something is done. This,
+   too, is best explained in [this section][linux-desc-changes] from the Linux
+   Kernel submitting patches guide.
+
+ * Commit titles should be meaningful and describe _what_ the commit does
+   ("refactor" and "fix" would be the title of ~all commits otherwise). For
+   example, when you are using "refactor" in a title, you probably include a list
+   of things in the commit message. Consider doing those things in different
+   commits instead. Commits should be atomic.
+
+ * Don't add code you will change later (it makes pointless to review that
+   commit), nor create a commit to add code an earlier commit should have
+   added. Consider to just squash the changes.
+
+ * Your development history is not important. Use git rebase to squash and order
+   commits in a way that is easy to review. Just keep the useful commits and not
+   your development history that led you to the change.
+
+ * Consider reviewing yourself the changes before opening a PR. Most of the
+   times, you can catch several errors.
+
+ * Use the PR description as a cover letter and give the context you think
+   people reviewing the code might need. Use the PR description to give an
+   overview, ask the questions about yet unsolved things in your PR, list
+   pending/TODO items, etc.
+
+If PRs follow these rules, usually, they are more easy to review and merge.
+
+It is totally okay to bend the rules sometimes. For example, if you reworked a
+piece of code and you realize it happens to fix other issues, that is totally
+fine. A strict interpretation of the "commits shouls be atomic" rule might make
+you think you need to fix the issues in the original code and _then_ do your
+rework that possibly deletes the fixes you created. That is not the idea, we
+don't want to create pointless overhead. Just do your rework and mention that it
+also fixes some other bugs that were present before your PR. Don't add code you
+will later delete (as another rule says).
```

Comment:
I would remove this paragraph. I think all it does is weaken the message we are trying to convey with these guidelines. People know guidelines aren't laws and we should trust them to use their good judgement when contributing.

Reply:
I wanted to emphasize that you shouldn't have to do a lot of extra work "to make your PR according to the rules". I thought that interpretation I used in the example was _plausible_ and wanted to make clear it was not the intention.

But if it clear for others too, maybe I'll remove it. Thanks! :)

Reply:
Okay, removed as @invidian said he agreed with the comments you did too :)